po/perform_translations/translator.py

```diff
@@ -0,0 +1,59 @@
+import re
+import dl_translate as dlt
+import polib
+
+# Variables for translator to edit
+
+language = "fr" # change to desired language to convert TO
+
+
+mt = dlt.TranslationModel()
+arrayOfTranslated = []
+msgIdArray = []
+msgStrArray = []
+pofile = polib.pofile(language + ".po") # Make sure the .po file is in the same directory
+
+for entry in pofile:
+    msgIdArray.append(entry.msgid)
+    msgStrArray.append(entry.msgstr)
+
+for messageId in msgIdArray:
+    reg = "|".join(
+        (
+            r"<\${\w+?}>",  # <${link}>
+            r"\${\w+?}",  # ${link}
+            r"<\$\w+?>",  # <$link>
+            r"\$\w+",  # $link
+            r"[{|}]",  # {a | b}
+        )
+    )
```

Comment:
This regular expression is not sufficient to cover all the syntax of template strings. The full syntax is at https://github.com/stanford-oval/genie-toolkit/blob/master/lib/utils/template-string/grammar.pegjs

In particular, I'm not sure how you're handling grammar flags, and phrases inside a ${select} clause.

Reply:
Ok yeah this makes sense. I suppose people will find a mix of English and target language and then deal with it. At least the see the original and can make sense of it.

Reply:
Yes, the msgid's are preserved.